core/sdk-core/src/main/java/software/amazon/awssdk/core/FileTransformerConfiguration.java

```diff
@@ -0,0 +1,209 @@
+ *
+ * @see #builder()
+ * @see FileWriteOption
+ * @see FailureBehavior
+ */
+@SdkPublicApi
+public final class FileTransformerConfiguration implements ToCopyableBuilder<FileTransformerConfiguration.Builder,
+    FileTransformerConfiguration> {
+    private final FileWriteOption fileWriteOption;
+    private final FailureBehavior failureBehavior;
+
+    private FileTransformerConfiguration(DefaultBuilder builder) {
+        this.fileWriteOption = Validate.paramNotNull(builder.fileWriteOption, "fileWriteOption");
+        this.failureBehavior = Validate.paramNotNull(builder.failureBehavior, "failureBehavior");
+    }
+
+    /**
+     * The configured {@link FileWriteOption}
+     */
+    public FileWriteOption fileWriteOption() {
+        return fileWriteOption;
+    }
+
+    /**
+     * The configured {@link FailureBehavior}
+     */
+    public FailureBehavior failureBehavior() {
+        return failureBehavior;
+    }
+
+    /**
+     * Create a {@link Builder}, used to create a {@link FileTransformerConfiguration}.
+     */
+    public static Builder builder() {
+        return new DefaultBuilder();
+    }
+
+    /**
+     * Returns the default {@link FileTransformerConfiguration} for {@link FileWriteOption#CREATE_NEW}
+     * <p>
+     * Always create a new file. If the file already exists, {@link FileAlreadyExistsException} will be thrown.
+     * In the event of an error, the SDK will attempt to delete the file (whatever has been written to it so far).
+     */
+    public static FileTransformerConfiguration defaultCreateNew() {
+        return builder().fileWriteOption(FileWriteOption.CREATE_NEW)
+                        .failureBehavior(FailureBehavior.DELETE)
+                        .build();
+    }
+
+    /**
+     * Returns the default {@link FileTransformerConfiguration} for {@link FileWriteOption#CREATE_OR_REPLACE_EXISTING}
+     * <p>
+     * Create a new file if it doesn't exist, otherwise replace the existing file.
+     * In the event of an error, the SDK will attempt to delete the file (whatever has been written to it so far).
+     */
+    public static FileTransformerConfiguration defaultCreateOrReplaceExisting() {
+        return builder().fileWriteOption(FileWriteOption.CREATE_OR_REPLACE_EXISTING)
+                        .failureBehavior(FailureBehavior.DELETE)
+                        .build();
+    }
+
+    /**
+     * Returns the default {@link FileTransformerConfiguration} for {@link FileWriteOption#CREATE_OR_APPEND_EXISTING}
+     * <p>
+     * Create a new file if it doesn't exist, otherwise append to the existing file.
+     * In the event of an error, the SDK will NOT attempt to delete the file, leaving it as-is
+     */
+    public static FileTransformerConfiguration defaultCreateOrAppend() {
+        return builder().fileWriteOption(FileWriteOption.CREATE_OR_APPEND_EXISTING)
+                        .failureBehavior(FailureBehavior.LEAVE)
```

Comment:
Any specific reason for this default?

Reply:
Yeah, retry would not work as expected if we delete the file when there is an exception.

Reply:
But it also might not work as expected if we started appending, failed and then on a retry reappended the same data.

Reply:
The initial position is recorded in the ctor, so we will always append on the same offset on retries.